crates/polars-mem-engine/src/executors/hive_scan.rs

```diff
@@ -0,0 +1,182 @@
+use std::borrow::Cow;
+
+use hive::HivePartitions;
+use polars_core::frame::column::ScalarColumn;
+use polars_core::utils::{
+    accumulate_dataframes_vertical, accumulate_dataframes_vertical_unchecked,
+};
+
+use super::Executor;
+use crate::executors::ParquetExec;
+use crate::prelude::*;
+
+pub trait ScanExec {
+    fn read(&mut self) -> PolarsResult<DataFrame>;
+}
+
+pub struct HiveExec {
+    sources: ScanSources,
```

Comment:
I think we can accept a generic here. One that take the filtered out paths and creates a `Box<dyn Executror>`. Then this node doesn't have to know anything about the actual implementations of parquet, csv, etc. The planner deals with that.